File: Java/EulerFunctions/PythagoreanTriplet.java
package ProjectEuler.EulerFunctions;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PythagoreanTriplet {

    public static void main(String[] args) {
        double start = System.nanoTime();

        System.out.println(pythagoreanTriplet(100));

        System.out.println((System.nanoTime() - start) / 1_000_000 + " ms");
    }

    private static List pythagoreanTriplet(int limit) {
        List triplets = new ArrayList<>();
        for (int a = 1; a < limit; a++) {
            for (int b = a + 1; b < limit; b++) {
                double c = Math.ceil(Math.sqrt(Math.pow(a, 2.0) + Math.pow(b, 2.0)));
                if (Math.pow(a, 2.0) + Math.pow(b, 2.0) == Math.pow(c, 2.0)) {
                    triplets.add(Arrays.toString(new int[]{a, b, (int) c}));
                }
            }
        }
        return triplets;
    }
}

File: Python/Euler Functions/Pythagorean Triplet.py
from time import clock

import math


def py_tri_v1(limit):
    triplets = []
    for a in range(1, limit):
        for b in range(1, limit):
            for c in range(1, limit):
                if a ** 2 + b ** 2 == c ** 2:
                    triplets.append([a, b, c])

    return triplets


def py_tri_v2(limit):
    triplets = []
    for a in range(1, limit):
        for b in range(a + 1, limit):
            for c in range(b + 1, limit):
                if a ** 2 + b ** 2 == c ** 2:
                    triplets.append([a, b, c])
    return triplets


def py_tri_v3(limit):
    triplets = []
    for a in range(1, limit):
        for b in range(a + 1, limit):
            c = math.ceil(math.sqrt(a ** 2 + b ** 2))
            if a ** 2 + b ** 2 == c ** 2:
                triplets.append([a, b, c])
    return triplets


start = clock()
print(py_tri_v3(100))
print(clock() - start)
